app/models/authenticator/httpd.rb

```diff
@@ -122,6 +130,20 @@ def user_details_from_external_directory(username)
     end
 
     def user_details_from_headers(username, request)
+      if debug_auth?
+        log_auth_debug(
+          [
```

Comment:
I have another suggestion here:

```ruby
  remote_headers = %w(X-REMOTE-USER X-REMOTE-USER-EMAIL X-REMOTE-USER-FIRSTNAME X-REMOTE-USER-LASTNAME X-REMOTE-USER-FULLNAME X-REMOTE-USER-GROUPS X-REMOTE-USER-DOMAIN)
  remote_headers.map do |rh|
    val = request.headers["#{rh}"]
    "request.headers[#{rh}] ->#{val}<-"
  end 
```

Using `username` instead of `request.headers['X-REMOTE-USER']` might mask a bug were username got corrupt somehow.

This approach might not work as well if the `CGI.unescape` is necessary for the groups. Maybe I'm missing something but I've never had to use it when dumping the groups.